Add missing required_tags to permission object and cli info

tmihalac opened the pull request:
- Add missing required_tags to permission object
- Added required_tags to cli info

Reference: [comments](https://github.com/feast-dev/feast/pull/4380#discussion_r1713941864)

dmartinol commented on docs/reference/feast-cli-commands.md:
```diff
@@ -205,12 +211,18 @@ types:
     roles:
     - reader
+tags:
+  key1: value1
```
could you please differentiate key names and values b/w `required_tags` and `tags`?

dmartinol commented on sdk/python/feast/permissions/permission.py:
```diff
@@ -196,7 +206,10 @@ def to_proto(self) -> PermissionProto:
             actions=actions,
             policy=self.policy.to_proto(),
-            tags=self._tags if self._tags is not None else None,
+            tags=self.tags if self.tags is not None else None,
```
isn't this equivalent to 
```py
tags=self.tags
required_tags=self.required_tags
```
In fact, when they are `None` the if command returns `None`

dmartinol reviewed (commented):
LGTM

tmihalac committed:
Add missing required_tags to permission object and cli info
- Add missing required_tags to permission object
- added required_tags to cli info

Signed-off-by: Theodor Mihalache <tmihalac@redhat.com>
